### PR TITLE
v2.2.0 added comment lines

### PR DESCRIPTION
**Pull request by germanysources**
Added the comment lines like in the mockup loader version 1.3.0

### CODE REVIEW

File: zcl_text2tab_parser.clas.abap [hunk 1]
```diff
@@ -56,6 +57,7 @@ private section.
   data MV_CURRENT_FIELD type STRING .
   data MV_LINE_INDEX type SY-TABIX .
   data MV_IS_TYPELESS type ABAP_BOOL .
+  data MV_BEGIN_COMMENT type RVARI_VAL_255 .
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.abap]: why so long value ? let's keep to 1 char.

File: zcl_text2tab_parser.clas.abap [hunk 2]
```diff
@@ -157,11 +159,16 @@ private section.
       !CODE type CHAR2 optional
     raising
       ZCX_TEXT2TAB_ERROR .
-  class-methods BREAK_TO_LINES
+  methods BREAK_TO_LINES
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.abap]: Let's keep static. Better for unit testing and less bug-prone, just pass comment char as a param

File: zcl_text2tab_parser.clas.abap [hunk 3]
```diff
@@ -157,11 +159,16 @@ private section.
...
     importing
       !I_TEXT type STRING
     returning
       value(RT_TAB) type STRING_TABLE .
+  methods IS_COMMENT_LINE
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.abap]: Same here, let's keep static

File: zcl_text2tab_parser.clas.abap [hunk 4]
```diff
@@ -204,7 +211,9 @@ endmethod.  "apply_conv_exit
 method BREAK_TO_LINES.
   data:
         l_found type i,
+        idx TYPE i,
         l_break type string value c_crlf.
+  FIELD-SYMBOLS: <line> TYPE string.
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.abap]: pls keep to formatting style of all other code. I mean lower cased key words. Same for all other code.
UPD: exception is the definitions, which are sometimes formatted on regeneration ...

File: zcl_text2tab_parser.clas.abap [hunk 5]
```diff
@@ -239,6 +255,18 @@ method CREATE.
     ro_parser->mv_date_format = i_date_format.
   endif.
 
+  CLEAR ro_parser->mv_begin_comment.
+  IF i_begin_comment IS SUPPLIED.
+    ro_parser->mv_begin_comment = i_begin_comment.
+  ELSE.
+    SELECT low UP TO 1 ROWS FROM tvarvc INTO ro_parser->mv_begin_comment
+      WHERE name = 'ZMOCKUP_LOADER_COMMENT' AND type = 'P'.
+    ENDSELECT.
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.abap]: let's do it in another PR. If tvarvc feature is implemented it should be for all params, not just for this one. Also when you do, could you pls use `select single`. And one question to consider - unit tests for tvarvc ... the selects should not affect unit tests. Let's discuss in a separate PR or issue pls.

File: zcl_text2tab_parser.clas.abap [hunk 6]
```diff
@@ -217,6 +226,13 @@ method BREAK_TO_LINES.
 
   split i_text at l_break into table rt_tab.
 
+  LOOP AT rt_tab ASSIGNING <line>.
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.abap]: `if mv_begin_comment is not initial` ??

File: zcl_text2tab_parser.clas.abap [hunk 7]
```diff
@@ -273,6 +301,20 @@ method GET_SAFE_STRUC_DESCR.
 endmethod.  "get_safe_struc_descr
 
 
+  method IS_COMMENT_LINE.
+    DATA: offset TYPE offset.
+
+    is = abap_false.
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.abap]: could you pls rename to `rv_***` ? (`rv_yes`, `rv_result`, `rv_is`)

File: zcl_text2tab_parser.clas.abap [hunk 8]
```diff
@@ -273,6 +301,20 @@ method GET_SAFE_STRUC_DESCR.
...
+
+    FIND FIRST OCCURRENCE OF REGEX '[^[:space:]]' IN line
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.abap]: I'd insist on checking strictly the first char of the string. There is no indentation in text data so it does not make sense to skip space. `*` can be a part of another field value

File: zcl_text2tab_parser.clas.testclasses.abap [hunk 9]
```diff
@@ -119,7 +119,8 @@ class lcl_text2tab_parser_test definition for testing
     methods parse_field_unsupp    for testing.
     methods map_head_structure    for testing.
     methods get_safe_struc_descr  for testing.
-    methods break_to_lines        for testing.
+    methods break_to_lines        for testing
+      RAISING zcx_text2tab_error.
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.testclasses.abap]: is raising needed ?

File: zcl_text2tab_parser.clas.xml [hunk 10]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (sbcgua) [inline zcl_text2tab_parser.clas.xml]: hmmm ... any idea why this looks changed ? :)

**Comment by germanysources** [comment]
I added the requested changes. 
There is now one unique char at the begin of the line for a comment.

**Comment by sbcgua** [comment]
OK, looks fine now, thanks
Just one more request: could you pls mention the feature in the documentation (readme.md). Maybe just add a section somewhere near "date formats"

P.S. btw for stvarv I created a separate issue: https://github.com/sbcgua/abap_data_parser/issues/5

**Comment by germanysources** [comment]
The documentation is added in the readme.

**Comment by sbcgua** [comment]
Nice! Merging then :tada:
